docs/source/notes/extending.rst

```diff
@@ -445,6 +445,43 @@ Also see the ``MetadataTensor`` example below for another variation on this
 pattern but instead always returns a ``MetadataTensor`` to propagate metadata
 through operations in the :mod:`torch` API.
 
+The ``__torch_function__`` protocol is designed for full coverage of the API,
+partial coverage may lead to undesirable results, in particular, certain
+functions raising a ``TypeError``. This is especially true for subclasses,
+where all three of `torch.add`, `torch.Tensor.__add__` and `torch.Tensor.add`
+must be covered, even if they return exactly the same result. Failing to do
+this may also lead to infinite recursion. If one requires the implementation
```

Comment:
Wait, is this ("infinite recursion") actually true? Do you have a simple example that triggers this?

Reply:
The example also existed in older versions for tensor-likes, now it's extended to methods as well.

Old version:

```python
In [5]: class TensorLike: 
   ...:     def __torch_function__(self, func, types, args=(), kwargs=None): 
   ...:         if kwargs is None: 
   ...:             kwargs = {} 
   ...:         return func(*args, **kwargs) 
   ...:                                                                                                                                                                                                                           

In [6]: torch.add(TensorLike(), TensorLike())                                                                                                                                                                                     
---------------------------------------------------------------------------
RecursionError                            Traceback (most recent call last)
<ipython-input-6-127f099dec10> in <module>
----> 1 torch.add(TensorLike(), TensorLike())

<ipython-input-5-c1183569170f> in __torch_function__(self, func, types, args, kwargs)
      3         if kwargs is None:
      4             kwargs = {}
----> 5         return func(*args, **kwargs)
      6 

... last 1 frames repeated, from the frame below ...

<ipython-input-5-c1183569170f> in __torch_function__(self, func, types, args, kwargs)
      3         if kwargs is None:
      4             kwargs = {}
----> 5         return func(*args, **kwargs)
      6 

RecursionError: maximum recursion depth exceeded while calling a Python object
```

New version:
```python
In [7]: class SubTensor(torch.Tensor): 
   ...:     def __torch_function__(self, func, types, args=(), kwargs=None): 
   ...:         if kwargs is None: 
   ...:             kwargs = {} 
   ...:         return func(*args, **kwargs) 
   ...:                                                                                                                                                                                                                           

In [8]: SubTensor([5]) + SubTensor([7])                                                                                                                                                                                           
---------------------------------------------------------------------------
RecursionError                            Traceback (most recent call last)
<ipython-input-8-cdb5550f0abb> in <module>
----> 1 SubTensor([5]) + SubTensor([7])

<ipython-input-7-c1229bb60ccd> in __torch_function__(self, func, types, args, kwargs)
      3         if kwargs is None:
      4             kwargs = {}
----> 5         return func(*args, **kwargs)
      6 

... last 1 frames repeated, from the frame below ...

<ipython-input-7-c1229bb60ccd> in __torch_function__(self, func, types, args, kwargs)
      3         if kwargs is None:
      4             kwargs = {}
----> 5         return func(*args, **kwargs)
      6 

RecursionError: maximum recursion depth exceeded
```

Reply:
I've added documentation to this effect.